common/compositor/gl/egloffscreencontext.cpp

```diff
@@ -102,22 +103,22 @@ void EGLOffScreenContext::RestoreState() {
 
 EGLint EGLOffScreenContext::GetSyncFD() {
   EGLint sync_fd = -1;
-#ifndef DISABLE_EXPLICIT_SYNC
-  EGLSyncKHR egl_sync =
-      eglCreateSyncKHR(egl_display_, EGL_SYNC_NATIVE_FENCE_ANDROID, NULL);
-  if (egl_sync == EGL_NO_SYNC_KHR) {
-    ETRACE("Failed to make sync object.");
-    return -1;
-  }
+  if (android::DrmHwcTwo::IsExplicitSyncEnabled()) {
+    EGLSyncKHR egl_sync =
```

Comment:
Instead of doing all this, what you can do is add a new API in https://github.com/01org/IA-Hardware-Composer/blob/master/public/nativedisplay.h void DisableExplicitSyncSupport()  and NativeDisplay will propogate it to compositor etc during creating time. DisableExplicitSyncSupport can be called from drmhwctwo during initialization phase depending on run time property.